tests/Tests.AzureAppConfiguration/PushRefreshTests.cs

```diff
@@ -0,0 +1,335 @@
+
+			PushNotification pushNotification = new PushNotification();
+
+			//Should returnFalse as empty or null string
+			Assert.False(EventGridEventParser.TryParseJson(emptyMessage, out pushNotification));
+			Assert.True(IsPushNotificationNull(pushNotification));
+			Assert.False(EventGridEventParser.TryParseJson(nullMessage, out pushNotification));
+			Assert.True(IsPushNotificationNull(pushNotification));
+
+			//Should return true since the parameter is put into PushNotification
+			//SDK will handle incorrect data/formatting in each parameter
+			Assert.True(EventGridEventParser.TryParseJson(badSyncTokenMsg, out pushNotification));
+			Assert.True(EventGridEventParser.TryParseJson(badUriMsg, out pushNotification));
+			Assert.True(EventGridEventParser.TryParseJson(badEventType, out pushNotification));
+			
+			//These should return false as parameter was not found and put into pushNotification
+			Assert.False(EventGridEventParser.TryParseJson(noSyncTokenMsg, out pushNotification));
+			Assert.True(IsPushNotificationNull(pushNotification));
+			Assert.False(EventGridEventParser.TryParseJson(noUriMsg, out pushNotification));
+			Assert.True(IsPushNotificationNull(pushNotification));
+			Assert.False(EventGridEventParser.TryParseJson(noEventTypeMsg, out pushNotification));
+			Assert.True(IsPushNotificationNull(pushNotification));
+		}
+
+		[Fact]
+		public void SyncTokenUpdatesCorrectNumberOfTimes()
+		{
+			// Arrange
+			var mockResponse = new Mock<Response>();
+			var mockClient = GetMockConfigurationClient();
+
+			IConfigurationRefresher refresher = null;
+
+			var config = new ConfigurationBuilder()
+				.AddAzureAppConfiguration(options =>
+				{
+					options.Client = mockClient.Object;
+					options.Select("*");
+					options.ConfigureRefresh(refreshOptions =>
+					{
+						refreshOptions.Register("TestKey1", "label")
+							.SetCacheExpiration(TimeSpan.FromDays(30));
+					});
+					refresher = options.GetRefresher();
+				})
+				.Build();
+
+			List<PushNotification> pushNotifications = CreatePushNotificationList();
+
+			foreach (PushNotification pushNotification in pushNotifications)
+			{
+				refresher.ProcessPushNotification(pushNotification, TimeSpan.FromSeconds(0));
+				refresher.RefreshAsync().Wait();
+			}
+
+			mockClient.Verify(c => c.GetConfigurationSettingAsync(It.IsAny<ConfigurationSetting>(), It.IsAny<bool>(), It.IsAny<CancellationToken>()), Times.Exactly(8));
+			mockClient.Verify(c => c.UpdateSyncToken(It.IsAny<string>()), Times.Exactly(8));
+		}
+
+		private bool IsPushNotificationNull(PushNotification pn)
```

Comment:
This method is not just checking if pushnotification is null. Its also checking if any of its properties are null. So I suggest renaming it to something like `IsPushNotificationInvalid`.

Reply:
On second thought, `IsPushNotificationInvalid` has negative semantics since it returns 'true' when something is 'invalid'. The better approach would be to use an affirmative phrase like `IsPushNotificationValid` and change the method logic accordingly.